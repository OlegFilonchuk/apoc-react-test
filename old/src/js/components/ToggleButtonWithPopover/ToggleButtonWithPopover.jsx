import React from 'react';
import PropTypes from 'prop-types';

import ToggleButton from '../ToggleButton/ToggleButton';
import withPopover from '../withPopover/withPopover';

const ToggleButtonComponent = withPopover(ToggleButton);

const ToggleButtonWithPopover = props => <ToggleButtonComponent {...props} />;

ToggleButtonWithPopover.propTypes = {
  tooltip: PropTypes.node.isRequired,
  ...ToggleButton.propTypes,
  ...ToggleButtonComponent.propTypes
};

export default ToggleButtonWithPopover;
