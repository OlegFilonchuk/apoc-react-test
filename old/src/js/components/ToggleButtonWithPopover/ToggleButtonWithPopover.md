Allow to render [`<ToggleButton>`](#togglebutton) together with [`<Popover>`](#popover) component. Thanks to that you can present user
tooltips when he is hovering cursor above the input field.

You can pass all of the [`<ToggleButton>` props](#togglebutton) to the `<ToggleButtonWithPopover>` component. 

# Example

  ```
  const initialState = {
    isOn: true,
    isDisabled: false
  };

  const changeState = () => {
    setState({
      isOn: !state.isOn
    });

    document.getElementById('toggleButtonWithPopoverConsole').innerText = `Now I'm ${!state.isOn ? 'ON': 'OFF'}`;
  };
      
  (
    <div>
      <ToggleButtonWithPopover
        tooltip="Some nice and helpfull text for the user" 
        isOn={state.isOn}
        onSwitch={changeState}
      />
      <div id="toggleButtonWithPopoverConsole"></div>
    </div>
  )
  ```
