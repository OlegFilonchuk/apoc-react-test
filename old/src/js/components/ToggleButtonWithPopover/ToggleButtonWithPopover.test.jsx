import React from 'react';
import { mount } from 'enzyme';

import ToggleButtonWithPopover from './ToggleButtonWithPopover';

describe('<ToggleButtonWithPopover />', () => {
  it('should render component', () => {
    const component = mount(<ToggleButtonWithPopover tooltip="Foo Bar tip" placeholder="What's up?" />);

    expect(component).toMatchSnapshot();
  });
});
