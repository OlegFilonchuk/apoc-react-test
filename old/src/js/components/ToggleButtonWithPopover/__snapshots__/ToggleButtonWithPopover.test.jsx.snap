// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleButtonWithPopover /> should render component 1`] = `
<ToggleButtonWithPopover
  placeholder="What's up?"
  tooltip="Foo Bar tip"
>
  <ToggleButtonWithPopover
    placeholder="What's up?"
    tooltip="Foo Bar tip"
  >
    <Popover
      className=""
      followCursorPosition={false}
      header={null}
      hideAfterDelayOf={500}
      hideArrow={false}
      hidePopup={false}
      isAllowed={true}
      isClosable={false}
      isDisabled={false}
      onClose={[Function]}
      onOutsidePopoverClick={[Function]}
      onPopoverClick={[Function]}
      onVisibilityChange={[Function]}
      openTrigger="mouseOver"
      placeholder="What's up?"
      placement="leftTop"
      popoverStyles={null}
      removeChildrenFromDOMWhenHidden={false}
      showAfterDelayOf={500}
      title={
        <ToggleButton
          buttonType="button"
          className={null}
          errorMessage={null}
          isDisabled={false}
          isLightTheme={false}
          isOn={true}
          labelOff="off"
          labelOn="on"
          onSwitch={[Function]}
          placeholder="What's up?"
          toggleId={null}
          toggleSize="lg"
        />
      }
      tooltipClassName=""
      type="default"
    >
      <div
        className="tooltip-container "
        onClick={[Function]}
      >
        <div
          className="popover-title"
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
        >
          <ToggleButton
            buttonType="button"
            className={null}
            errorMessage={null}
            isDisabled={false}
            isLightTheme={false}
            isOn={true}
            labelOff="off"
            labelOn="on"
            onSwitch={[Function]}
            placeholder="What's up?"
            toggleId={null}
            toggleSize="lg"
          >
            <div
              className="toggle-button lg is-on"
              id={null}
              onClick={[Function]}
            >
              <div
                className="items-tumbler on"
              >
                <div
                  className="label"
                  title="on"
                >
                  on
                </div>
                <button
                  className="big-toggle__inner-circle"
                  type="button"
                />
                <div
                  className="label"
                  title="off"
                >
                  off
                </div>
              </div>
            </div>
          </ToggleButton>
        </div>
        <div
          className="qtip qtip-chameleon qtip-chameleon-default qtip-shadow qtip-pos-lt qtip-focus"
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          style={
            Object {
              "display": "none",
            }
          }
        >
          <div
            className="qtip-content"
            onClick={[Function]}
          >
            <div
              className="visible"
            >
              <header />
              <div
                className="tooltip-body"
              >
                Foo Bar tip
              </div>
            </div>
          </div>
        </div>
      </div>
    </Popover>
  </ToggleButtonWithPopover>
</ToggleButtonWithPopover>
`;
